src/api/app/controllers/webui/user_controller.rb

```diff
@@ -156,6 +156,11 @@ def tokens
     render json: User.autocomplete_token(params[:q])
   end
 
+  def acknowledge_announcement
```

Comment:
Hm does it not make more sense to move it to announcements_controller as :update action? 
(Also because the user controller is already really heavy)

Reply:
@ChrisBr  I think the best place could be `AnnouncementsUsersController` because we are creating an association, I mean, adding a new entry to `announcements_users` table. But we don't have this controller and don't think is useful to create it.

So, choosing between User or Announcement, I think it has more sense like "A user accepts/acknowledges an announcement" instead of "An announcement is accepted/acknowledged by a user".

Reply:
> But we don't have this controller and don't think is useful to create it.

Why not?

Reply:
> So, choosing between User or Announcement, I think it has more sense like "A user accepts/acknowledges an announcement" instead of "An announcement is accepted/acknowledged by a user".

That makes sense! 

>  I think the best place could be AnnouncementsUsersController

Why not following what we do on many other controllers and introduce a nested controller like
```
User::AnnouncementsController
```

http://jeromedalbert.com/how-dhh-organizes-his-rails-controllers/